### PR TITLE
Build manifest for org.mozilla.Firefox

### PR DESCRIPTION
**Pull request by nhlfr**

### CODE REVIEW

File: org.mozilla.Firefox.json [hunk 1]
```diff
@@ -0,0 +1,136 @@
+{
+  "app-id": "org.mozilla.Firefox",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "3.22",
+  "sdk": "org.gnome.Sdk",
+  "command": "firefox",
+  "separate-locales": false,
```
Comment (TingPing) [inline org.mozilla.Firefox.json]: WhY?

File: org.mozilla.Firefox.json [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  "finish-args": [
+    "--share=ipc",
+    "--socket=wayland",
```
Comment (TingPing) [inline org.mozilla.Firefox.json]: It doesn't support Wayland yet.

File: org.mozilla.Firefox.json [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    "--socket=x11",
+    "--socket=pulseaudio",
+    "--share=network",
+    "--filesystem=home"
+  ],
+  "modules": [
+    {
+      "name": "autoconf",
```
Comment (TingPing) [inline org.mozilla.Firefox.json]: In Sdk..

File: org.mozilla.Firefox.json [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://github.com/webmproject/libvpx/archive/v1.6.0.tar.gz",
+          "sha256": "e2fc00c9f60c76f91a1cde16a2356e33a45b76a5a5a1370df65fd57052a4994a"
+        }
+      ]
+    },
+    {
+      "name": "gstreamer",
```
Comment (TingPing) [inline org.mozilla.Firefox.json]: In Sdk...

File: org.mozilla.Firefox.json [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://gstreamer.freedesktop.org/src/gstreamer/gstreamer-1.10.2.tar.xz",
+          "sha256": "150e8e81febac94c161d8141cde78a38038a8f56e8ec549f353da54994278d65"
+        }
+      ]
+    },
+    {
+      "name": "gst-plugins-base",
```
Comment (TingPing) [inline org.mozilla.Firefox.json]: In Sdk..

File: org.mozilla.Firefox.desktop [hunk 6]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
```
Comment (TingPing) [inline org.mozilla.Firefox.desktop]: Does it not come with a desktop file?

File: org.mozilla.Firefox.json [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+            "--enable-nls",
+            "--with-ssl=openssl"
+          ],
+          "url": "https://download.gnome.org/sources/GConf/3.2/GConf-3.2.6.tar.xz",
+          "sha256": "1912b91803ab09a5eed34d364bf09fe3a2a9c96751fde03a4e0cfa51a04d784c"
+        }
+      ]
+    },
+    {
+      "name": "yasm",
```
Comment (TingPing) [inline org.mozilla.Firefox.json]: In Sdk..

File: org.mozilla.Firefox.json [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+      "sources": [
+        {
+          "type": "archive",
+          "url": "http://www.tortall.net/projects/yasm/releases/yasm-1.3.0.tar.gz",
+          "sha256": "3dce6601b495f5b3d45b59f7d2492a340ee7e84b5beca17e48f862502bd5603f"
+        }
+      ]
+    },
+    {
+      "name": "libvpx",
```
Comment (TingPing) [inline org.mozilla.Firefox.json]: In Sdk..

**Comment by Barthalion** [comment]
@alexlarsson I believe that you said at least once that we don't want apps like Firefox on Flathub yet due to limited build infra. Is that still the case? I'm especially asking because we already have some rather big games merged and apps incoming (like darktable).

File: org.mozilla.Firefox.appdata.xml [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  <summary>Browse the Web</summary>
+  <description>
+    <p>
+    Bringing together all kinds of awesomeness to make browsing better for you. Get to your favorite sites quickly – even if you don’t remember the URLs. Type your term into the location bar (aka the Awesome Bar) and the autocomplete function will include possible matches from your browsing history, bookmarked sites and open tabs.
+    </p>
+  </description>
+  <url type="homepage">https://www.mozilla.org/en-US/</url>
+  <screenshots>
+    <screenshot type="default">
+      <image>http://www.codeblocks.org/images/stories/imagebrowser/scr1.png</image>
```
Comment (Barthalion) [inline org.mozilla.Firefox.appdata.xml]: CodeBlocks screenshot.

File: org.mozilla.Firefox.json [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+        "--disable-static"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://gstreamer.freedesktop.org/src/gst-plugins-base/gst-plugins-base-1.10.2.tar.xz",
+          "sha256": "fbc0d40fcb746d2efe2ea47444674029912f66e6107f232766d33b722b97de20"
+        }
+      ]
+    },
```
Comment (Barthalion) [inline org.mozilla.Firefox.json]: Everything that's not from Sdk should have some `cleanup` directives. We already have some cleaning up to do, we shouldn't increase that debt.

File: org.mozilla.Firefox.json (hunk 3 above)
Comment (nhlfr) [inline org.mozilla.Firefox.json]: Unfortunately, Firefox supports only 2.13 version of autoconf and Sdk has some newer. Using newer autoconf results in the following error:

```
========================================================================
Building module firefox in /home/vador/repos/firefox-flatpak/.flatpak-builder/build/firefox-2
========================================================================
Klone nach '/home/vador/repos/firefox-flatpak/.flatpak-builder/build/firefox-2' ...
Fertig.
Checke Dateien aus: 100% (185885/185885), Fertig.
Branch esr45 konfiguriert zum Folgen von Remote-Branch esr45 von origin.
Zu neuem Branch 'esr45' gewechselt
Applying patch firefox-makeinstall.patch
patching file GNUmakefile
Applying patch firefox-extension-dir.patch
patching file xpcom/io/nsAppFileLocationProvider.cpp
Hunk #1 succeeded at 197 (offset 10 lines).
./mach build
 0:00.25 /usr/bin/make -f client.mk -s
 0:00.26 make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
 0:00.54 client.mk:202: /run/build/firefox/obj-x86_64-unknown-linux-gnu/.mozconfig.mk: No such file or directory
 0:00.81 Clobber not needed.
 0:00.82 make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
 0:01.62 Adding client.mk options from /run/build/firefox/.mozconfig:
 0:01.62     MOZ_OBJDIR=/run/build/firefox/obj-x86_64-unknown-linux-gnu
 0:01.62     OBJDIR=/run/build/firefox/obj-x86_64-unknown-linux-gnu
 0:01.62     FOUND_MOZCONFIG=/run/build/firefox/.mozconfig
 0:01.63 /run/build/firefox/client.mk:299: *** Could not find autoconf 2.13.  Stop.
 0:01.63 client.mk:171: recipe for target 'build' failed
 0:01.63 make[1]: *** [build] Error 2
 0:01.67 0 compiler warnings present.
 0:02.68 ccache (direct) hit rate: 0.0%; (preprocessed) hit rate: 0.0%; miss rate: 100.0%
GNUmakefile:9: recipe for target 'build' failed
make: *** [build] Error 2
Error: module firefox: Der Kindprozess wurde mit Status 2 beendet
Makefile:9: die Regel für Ziel „test“ scheiterte
make: *** [test] Fehler 1
```
I can try to patch Firefox and bring the support for the latest autoconf it that sounds better for you.

File: org.mozilla.Firefox.desktop (hunk 6 above)
Comment (nhlfr) [inline org.mozilla.Firefox.desktop]: Unfortunately not.

File: org.mozilla.Firefox.json (hunk 7 above)
Comment (nhlfr) [inline org.mozilla.Firefox.json]:
```
 0:08.29 configure: error: Yasm is required to build with libav's optimized FFT routines, but you do not appear to have Yasm installed. See https://developer.mozilla.org/en/YASM for more details.
 0:08.29 *** Fix above errors and then restart with\
 0:08.29                "/usr/bin/make -f client.mk build"
 0:08.29 /run/build/firefox/client.mk:359: recipe for target 'configure' failed
 0:08.29 make[3]: *** [configure] Error 1
 0:08.29 /run/build/firefox/client.mk:373: recipe for target '/run/build/firefox/obj-x86_64-unknown-linux-gnu/Makefile' failed
 0:08.29 make[2]: *** [/run/build/firefox/obj-x86_64-unknown-linux-gnu/Makefile] Error 2
 0:08.29 client.mk:171: recipe for target 'build' failed
 0:08.29 make[1]: *** [build] Error 2
 0:08.32 0 compiler warnings present.
 0:09.33 ccache (direct) hit rate: 3.3%; (preprocessed) hit rate: 76.7%; miss rate: 20.0%
GNUmakefile:9: recipe for target 'build' failed
make: *** [build] Error 2
Error: module firefox: Der Kindprozess wurde mit Status 2 beendet
Makefile:9: die Regel für Ziel „test“ scheiterte
make: *** [test] Fehler 1
```
Looking at that, maybe we will need an another patch...

**Comment by alexlarsson** [comment]
I don't think we want firefox in flathub, but not because of the build performance, but rather that mozilla themselves are working on an official upstream flatpak release, and firefox is very sensitive to security updates, etc, which we'd rather they handle.

**Comment by nhlfr** [comment]
@TingPing I think we have a bigger problem with yasm.

When I removed yasm from manifest, then it isn't available at all:

```
$ flatpak-builder --run firefox org.mozilla.Firefox.json sh
sh-4.3$ which yasm
which: no yasm in (/app/bin:/usr/bin)
```

which probably means that GNOME Sdk doesn't have it.

The same command of course works in freedesktop-based flatpak:

```
$ flatpak-builder --run vlc org.videolan.VLC.json sh
sh-4.3$ which yasm
/usr/bin/yasm
```

This looks strange, because GNOME Sdk image is based on freedesktop
https://github.com/GNOME/gnome-sdk-images/blob/master/org.gnome.Sdk.json.in#L7

I will close this PR due to what @alexlarsson wrote, but I think we need to look at the issue with yasm in GNOME runtime.

**Comment by thiblahute** [comment]
> I don't think we want firefox in flathub, but not because of the build performance, but rather that mozilla themselves are working on an official upstream flatpak release, and firefox is very sensitive to security updates, etc, which we'd rather they handle.

Is it still the case?

fwiw I did https://github.com/thiblahute/org.mozilla.Firefox for myself where I just use their builds, and is much nicer to user than their tarball (proper integration in the system).

**Comment by TingPing** [comment]
Nothing has changed. Also Firefox is now starting to get real sandboxing which will be disabled under Flatpak (until it supports some form of recursive sandboxing) which decreases the security for users.

**Comment by jaduncan** [comment]
Does the sandboxing issue mean that the Mozilla upstream package is never coming?

**Comment by TingPing** [comment]
@jaduncan 	It may come but it won't be a third party package, we want Mozilla to build official Flatpaks.

**Pull request closed without merging**